From 672d016006da97c5a4c70a429b7f57bef9725e48 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Thu, 12 Sep 2013 23:53:59 -0400
Subject: [PATCH] Vold: Do not mount ASEC containers if moving apps to SD is
 disabled

Thanks to Pierre-Hugues Husson for the idea in this patch! The initial
implementation called createBindMounts() on all mount points regardless
of whether or not it was the SD card.

Change-Id: I85029272e8cc33d2b0044b5e4d05e20ba55db12d
---
 Volume.cpp | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/Volume.cpp b/Volume.cpp
index d3e627c..563437e 100644
--- a/Volume.cpp
+++ b/Volume.cpp
@@ -241,7 +241,16 @@ bool Volume::isPrimaryStorage() {
 
 bool Volume::isExternalSd() {
     const char* externalSd = getenv("EXTERNAL_SD") ? : "/storage/sdcard1";
-    return !strcmp(getMountpoint(), externalSd);
+    const char* disablePath = "/data/system/no-external-apps";
+
+    if (!strcmp(getMountpoint(), externalSd)) {
+        if (access(disablePath, F_OK) != -1) {
+            SLOGV("Application moving disabled; will not touch ASEC\n");
+            return false;
+        }
+        return true;
+    }
+    return false;
 }
 
 void Volume::setLunNumber(int lunNumber) {
-- 
1.8.4

